tests/popup-test.js

```diff
@@ -1,22 +1,118 @@
+ * distributed under the License is distributed on an 'AS-IS' BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+const popup = require('../readiness-tool/popup');
 
-describe('test', function() {
+const vendors = require('../readiness-tool/vendors');
 
-   it('should pass', function() {
-     expect(1).to.equal(1);
-   });
- });
+const Validator = require('jsonschema').Validator;
+
+const v = new Validator();
+
+describe('isSupported(key)', function () {
+
+  it('should return true when the key given is in the supported list', function () {
+    expect(popup.isSupported('Adhese')).to.be.true;
+  });
+
+  it('should return false when the key given is in NOT in the supported list', function () {
+    expect(popup.isSupported('alannalytics')).to.be.false;
+  });
+});
+
+describe('addToDict(tempScript, htmlString, foundThis, key, category)', function () {
+
+  let htmlString;
+  let foundThis;
+
+  beforeEach(() => {
+    htmlString = 'candycanes';
+    foundThis = {
+      'supported': {
+        'ads': [],
+        'analytics': []
+      },
+      'notSupported': {
+        'ads': [],
+        'analytics': []
+      }
+    };
+  });
+
+  it('should push analytics to the analytics array', function () {
+    let tempScript = 'candy';
+    const category = 'Analytics';
+    let key = 'comScore';
+    popup.addToDict(tempScript, htmlString, foundThis, key, category);
+    expect(foundThis.supported.analytics).to.include('comScore');
+    tempScript = 'peppermint';
+    expect(foundThis.supported.analytics).to.not.include('Swoop');
+    expect(foundThis.notSupported.analytics).to.not.include('Swoop');
+  });
+
+  it('should push ads to the ads array', function () {
+    let tempScript = 'candy';
+    const category = 'Ads';
+    let key = 'comScore';
+    popup.addToDict(tempScript, htmlString, foundThis, key, category);
+    expect(foundThis.supported.ads).to.include('comScore');
+    tempScript = 'peppermint';
+    expect(foundThis.supported.ads).to.not.include('Swoop');
+    expect(foundThis.notSupported.ads).to.not.include('Swoop');
+  });
+});
+
+describe('vendors.json should be valid json', function () {
+
+  it('should be a valid file', function () {
+    
+    let isJSONValid;
+    
+    const vendorSchema = {
+      'id': '/SimpleVendors',
+      'type': 'object',
+      'properties': {
+        'regex': {
+          'type': 'array',
+          'items': {
+            'type': 'string'
+          }
+        },
+        'tooltip': {
+          'type': 'string'
+        },
+        'category': {
+          'type': 'string'
+        },
+      },
+      'required': ['category', 'regex']
+    };
+
+    v.addSchema(vendorSchema, '/SimpleVendors');
+    
+    const result = v.validate(vendors.vendor, vendorSchema);
+    const jsonObj = JSON.parse(JSON.stringify(vendors));
```

Comment:
If there's a bug, these methods will probably throw an error before your assert below. The test will still fail anyways though, so this might be fine.